### PR TITLE
Add tests for `find`/`getMore` `batchSize`

### LINKED ISSUE #2005: Improve `find`/`getMore` `batchSize`/`singleBatch` handling
### What should be done?

Those methods handle corner cases (unset `batchSize`, 0 value, float value, etc) differently. See https://www.mongodb.com/community/forums/t/mis-understanding-batchsize/169713/2

We should test all combinations.

DOD:
- [ ] `batchSize` accepts float value.
- [ ] cursor returns number of documents set by `batchSize`.
- [ ] when `batchSize` is unset, cursor returns default `batchSize` 101 documents in `firstBatch`.
- [ ] when `batchSize` is unset, cursor returns remaining documents in `secondBatch`.
- [ ] calling `getMore` using `cursorID` gets more documents.
- [x] driver sets  `singleBatch` for negative `limit(-1)` on `find`, no matter the `batchSize`
- [ ] See issue number added in https://github.com/FerretDB/FerretDB/pull/2825

### PR DESCRIPTION
**Pull request by chilagrow**
# Description

Closes #2005.

## Readiness checklist

- [ ] I added/updated unit tests.
- [x] I added/updated integration/compatibility tests.
- [ ] I added/updated comments and checked rendering.
- [ ] I made spot refactorings.
- [ ] I updated user documentation.
- [ ] I ran `task all`, and it passed.
- [x] I ensured that PR title is good enough for the changelog.
- [x] (for maintainers only) I set Reviewers ([`@FerretDB/core`](https://github.com/orgs/FerretDB/teams/core)), Labels, Project and project's Sprint fields.
- [x] I marked all done items in this checklist.

### CODE REVIEW

File: integration/query_test.go [hunk 1]
```diff
@@ -613,3 +614,211 @@ func TestQueryNonExistingCollection(t *testing.T) {
...
+
+			firstBatch, ok := cursor.Map()["firstBatch"]
+			require.True(t, ok)
+			require.Equal(t, tc.firstBatch, firstBatch)
+		})
+	}
+}
+
+func TestQueryBatchSize(t *testing.T) {
+	t.Skip("https://github.com/FerretDB/FerretDB/issues/2005")
```
Comment (chilagrow) [inline integration/query_test.go:742]: Tests are currently skipped, but if we prefer I can make it not skipped on MongoDB only

File: integration/query_test.go [hunk 2]
```diff
@@ -613,3 +614,211 @@ func TestQueryNonExistingCollection(t *testing.T) {
...
+		err = cursor.All(ctx, &res)
+		require.NoError(t, err)
+
+		// cursor is exhausted
+		ok = cursor.Next(ctx)
+		require.False(t, ok, "batchSize has reached, not expecting next document")
+	})
+
+	t.Run("DefaultBatchSize", func(t *testing.T) {
+		// set BatchSize to 2
```
Comment (w84thesun) [inline integration/query_test.go:801]: Is this comment correct?

**Review by w84thesun — COMMENTED** [review]
Overall good, added a small question about the comment.

File: integration/query_test.go (hunk 2 above)
Comment (chilagrow) [reply integration/query_test.go]: Not correct :see_no_evil: Fixed

File: integration/helpers.go [hunk 3]
```diff
@@ -405,3 +405,14 @@ func FindAll(t testing.TB, ctx context.Context, collection *mongo.Collection) []
 
 	return FetchAll(t, ctx, cursor)
 }
+
+// generateDocuments generates documents with _id ranging from startID to endID.
+// It returns bson.A containing bson.D documents.
+func generateDocuments(startID, endID int32) bson.A {
```
Comment (chilagrow) [inline integration/helpers.go:411]: This is similar to what was done in https://github.com/FerretDB/FerretDB/pull/1769/files#diff-8307cfe62efa2f64d4ef2d0aff407a8af0986af9f9efe7f98de217e13fe89297R390

**Review by rumyantseva — APPROVED** [review]
The test cases look reasonable to me! I expect we might need to test some more things in the future, but it's hard to predict now, so it looks good for the beginning!

**Review by w84thesun — APPROVED** [review]
LGTM!

**Pull request merged**